### ISSUE TITLE
Docker thinks Windows containers are running after a reboot

### ISSUE THREAD

**Issue by paulb2015** [docker/for-win#4705]
Docker version: 18.09.3

After a host rebooted unexpectedly on Azure, Docker reported that the containers were still running with a status of days/weeks even though the VM had just rebooted. The containers have a restart policy of always.

Restarting the Docker Service or rebooting the VM again made no difference to the state of Docker.

I ran docker ps after the Docker Service had started:

```
C:\>docker ps
CONTAINER ID        IMAGE               COMMAND                    CREATED             STATUS              PORTS               NAMES
959ac84c60ab        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   13 days ago         Up 13 days                              app-10
c7581bbf6705        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   13 days ago         Up 13 days                              app-09
38bc73627650        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   13 days ago         Up 13 days                              app-08
6757c45489ea        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   13 days ago         Up 13 days                              app-07
ef87fde1c1b6        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   13 days ago         Up 13 days                              app-06
628c793d435f        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   2 weeks ago         Up 2 weeks                              app-05
05dbc9e77e86        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   2 weeks ago         Up 2 weeks                              app-04
4f168dc7fba9        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   2 weeks ago         Up 2 weeks                              app-03
869544b0ef0a        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   2 weeks ago         Up 2 weeks                              app-02
3b31bc1b5140        XXXXXXXXXXXXXXXX    "c:\\bootstrap\\bootst…"   2 weeks ago         Up 2 weeks                              app-01
```

Attempted to access the containers:

```
C:\>docker exec -t -i app-01 cmd.exe
no such container

C:\>docker exec -t -i app-02 cmd.exe
no such container

C:\>docker exec -t -i app-03 cmd.exe
no such container

C:\>docker exec -t -i app-04 cmd.exe
no such container

C:\>docker exec -t -i app-05 cmd.exe
no such container

C:\>docker exec -t -i app-06 cmd.exe
no such container

C:\>docker exec -t -i app-07 cmd.exe
no such container

C:\>docker exec -t -i app-08 cmd.exe
no such container

C:\>docker exec -t -i app-09 cmd.exe
no such container

C:\>docker exec -t -i app-10 cmd.exe
no such container

C:\>
```

The event viewer reported the following on start-up:

```
Failed to delete container 05dbc9e77e865e569be5f3f5de27db8e54f800ad4af0e107e44528df9600bd01 from containerd [error=no such container]
Failed to delete container 959ac84c60ab893c6a11f664dd4fb450a60c770a56fe50c0f64c3bb61c1e8161 from containerd [error=no such container]
Failed to delete container 628c793d435fa078821091a8a7eaaf12d4919bf8942a509fdabb6e5605a17c57 from containerd [error=no such container]
Failed to delete container ef87fde1c1b6f6f7571e5edb432ab6583bdea0c7a008e26adc465030f3cdf068 from containerd [error=no such container]
Failed to delete container 4f168dc7fba93517a25be804001c2df902aa5deb5637ec37e29bbdac3f8ee879 from containerd [error=no such container]
Failed to delete container 6757c45489eaac51c2594029963f22908b1eb06b18fde97ade05e7887603879f from containerd [error=no such container]
Failed to delete container 3b31bc1b51405943d50bd1888d92af11cacb7179fc8692034aadee7a3551cad2 from containerd [error=no such container]
Failed to delete container c7581bbf6705a5b5d68ec8fea94602d38f34c15f713d8cd4a4a4b83d28a94d06 from containerd [error=no such container]
Failed to delete container 38bc7362765051db746fca908ede8ae2233a350b3982fe0f37328366815df9cc from containerd [error=no such container]
Failed to delete container 869544b0ef0a2d4414e5ea947152cb7188b5a271492c70b250ef4c6f22802b49 from containerd [error=no such container]
```

I then installed Docker 19.03.2 on the VM, and ran docker ps, it showed that the containers were now running propertly (status time was seconds instead of weeks and I could access the container).

However I am not certain if this problem has been fixed in the later version of Docker or if a re-install of Docker has simply repaired something that was broken (i.e. I should have tried re-installing the same version first).

**Comment by jasoncoding** [comment]
Can confirm I saw this exact same behavior at nearly the same day.  Also on azure... wonder if I was in the same wave of patch/restart as your vm, paulb2015.

I tried everything my limited docker skills provide and couldn't remove the zombie containers until I completely reinstalled docker.  Which was interesting because the normal docker desktop for windows installed but then had an error "unsupported version".  FWIW, the install process I ended up using was a powershell cli: `Install-Package -Name docker -ProviderName DockerMsftProvider -Force -RequiredVersion 19.03.2`
